Split up testing IAM policies and automate creating them

willthames opened the pull request:
##### SUMMARY

Move to managed policies to avoid the 5KB limit on policies
for an IAM entity.

The policy file is templated, so need to make sure that there
is an easy mechanism to populate the templates and push the
new policies.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
hacking/aws_config

##### ANSIBLE VERSION
```
ansible 2.4.0 (devel dcd1ff2809) last updated 2017/06/29 13:11:02 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/will/src/ansible/lib/ansible
  executable location = /home/will/src/ansible/bin/ansible
  python version = 2.7.13 (default, May 10 2017, 20:04:28) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
```

Commit:
Split up testing IAM policies and automate creating them

Move to managed policies to avoid the 5KB limit on policies
for an IAM entity.

The policy file is templated, so need to make sure that there
is an easy mechanism to populate the templates and push the
new policies.

willthames commented:
Relies on #26219 for change of `PolicyName` to `policy_name`.

willthames commented:
cc @mattclay @mikedlr @shertel 
This should improve manageability of IAM for CI. At the moment it *should* create no permissions change from before.

mattclay commented:
@willthames Thanks for splitting these out. It should make things easier for manual testing. These policies aren't used for CI, so there won't be any impact there.

Commit:
Update IAM policies for ec2_scaling_policy tests

Commit:
Fix RouteTable policies

DescribeRouteTable should be plural
ModifyRouteTable does not exist, but ReplaceRouteTableAssociation
does.

Commit:
Some IAM policies do not allow specified Resources

Various IAM policies do not allow Resources
to be specified and should just use `*`. This differs
per service

* [Autoscaling](http://docs.aws.amazon.com/autoscaling/latest/userguide/control-access-using-iam.html#policy-auto-scaling-resources)
* [EC2](http://docs.aws.amazon.com/AWSEC2/latest/APIReference/ec2-api-permissions.html#ec2-api-unsupported-resource-permissions)
* [ECR](http://docs.aws.amazon.com/AmazonECR/latest/userguide/ecr-supported-iam-actions-resources.html)
* [ELB](http://docs.aws.amazon.com/elasticloadbalancing/latest/userguide/load-balancer-authentication-access-control.html)

Commit:
Finish fixing AWS IAM resource specifications for testing

Update Lambda and RDS policies
